### PR TITLE
USB extra details, GNU/Linux keystrokes process, TLS brief handshake process added

### PR DESCRIPTION
**Pull request by alx741**
- Corrected, and extra details for USB keyboards keycodes fetching.
- GNU/Linux graphical enviroment (X server, window maganer) to actual focused window process.
- TLS handshake reduced process explaination.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -92,6 +97,18 @@ handled by, an ``NSApplication`` main event loop, via an ``NSEvent`` of
 ``NSEventType`` ``KeyDown``.
 
 
+(On GNU/Linux) the Xorg server listen for keycodes
+--------------------------------------------------
+
+When a graphical ``X server`` is used, a re-maping of keycodes to scancodes
```
Comment (phildini) [inline README.rst]: re-mapping, I think is the word you're looking for

File: README.rst [hunk 2]
```diff
@@ -92,6 +97,18 @@ handled by, an ``NSApplication`` main event loop, via an ``NSEvent`` of
...
+is made with ``X server`` specific keymaps and rules.
+When the scancode maping of the key pressed is complete, the ``X server``
```
Comment (phildini) [inline README.rst]: mapping

File: README.rst [hunk 3]
```diff
@@ -92,6 +97,18 @@ handled by, an ``NSApplication`` main event loop, via an ``NSEvent`` of
...
+sends the character to the ``window manager`` (DWM, metacity, i3, etc), so the
+``window manager`` in turn sends the character to the focused window.
+The graphical API of the window  that receives the character prints the appropiate
```
Comment (phildini) [inline README.rst]: appropriate

File: README.rst [hunk 4]
```diff
@@ -187,6 +204,27 @@ UDP packets
 
 TLS handshake...
 ----------------
+* The client computer sends a ``Client hello`` message to the server with it TLS version,
+list of cipher algorithms available and compression methods availabe.
```
Comment (phildini) [inline README.rst]: available

File: README.rst [hunk 5]
```diff
@@ -187,6 +204,27 @@ UDP packets
...
+
+* The server replies with a ``Server hello`` message to the client with the TLS version, cipher
+and compression methods selected + the Server public certificate signed by a CA (Certificate Authority)
+that also contains a public key.
+
+* The client verifies the server digital certificate and cipher a simetric cryptography key using a asimetric
```
Comment (phildini) [inline README.rst]: symmetric, asymmetric

File: README.rst [hunk 6]
```diff
@@ -187,6 +204,27 @@ UDP packets
...
+  cryptography algorithm and the server public key and an encrypted message for verification purposes.
+
+* The server decrypt the key using its private key and decrypt the verification message with it, then replies
```
Comment (phildini) [inline README.rst]: decrypts, in both cases

**Comment by alx741** [comment]
Sorry for that mess, i guess is better now

**Comment by alx741** [comment]
I've screw it up, closing the pull request and trying again, apologize

**Pull request closed without merging**